src/models/flinkStatement.ts

```diff
@@ -0,0 +1,55 @@
+import { ThemeIcon, TreeItem, TreeItemCollapsibleState } from "vscode";
+import { ConnectionType } from "../clients/sidecar";
+import { IconNames } from "../constants";
+import { ConnectionId, EnvironmentId, IResourceBase, ISearchable } from "./resource";
+
+export class FlinkStatement implements IResourceBase, ISearchable {
+  connectionId!: ConnectionId;
+  connectionType!: ConnectionType;
+  iconName: IconNames = IconNames.FLINK_STATEMENT;
+
+  environmentId!: EnvironmentId;
+  computePoolId!: string;
+
+  name!: string;
+  status!: string;
+
+  // TODO: add more properties as needed
+
+  constructor(
+    props: Pick<
+      FlinkStatement,
+      "connectionId" | "connectionType" | "environmentId" | "computePoolId" | "name" | "status"
+    >,
+  ) {
+    this.connectionId = props.connectionId;
+    this.connectionType = props.connectionType;
+    this.environmentId = props.environmentId;
+    this.computePoolId = props.computePoolId;
+    this.name = props.name;
+    this.status = props.status;
+  }
+
+  searchableText(): string {
+    return `${this.name} ${this.status}`;
+  }
+}
+
+export class FlinkStatementTreeItem extends TreeItem {
+  resource: FlinkStatement;
+
+  constructor(resource: FlinkStatement) {
+    super(resource.name, TreeItemCollapsibleState.None);
+
+    // internal properties
+    this.id = `${resource.connectionId}-${resource.computePoolId}-${resource.name}`;
```

Comment:
Suggest then going with `resource.id`. We oughtn't ever be comparing / viewing `FlinkStatementTreeItems`  from multiple environments in the same view.